samples/sync-history/src/main/java/org/mozilla/samples/sync/history/MainActivity.kt

```diff
@@ -85,18 +85,28 @@ class MainActivity : AppCompatActivity(), LoginFragment.OnLoginCompleteListener,
             txtView.text = getString(R.string.logged_out)
         }
 
+        featureSync.onSyncRunning(this, object : onSyncRunning { override fun invoke() {
+            launch {
+                val txtView: TextView = findViewById(R.id.historySyncStatus)
+                txtView.text = getString(R.string.syncing)
+            }
+        } })
```

Comment:
Since this is a class with just one method you can write it as a lambda. And then you can move the lambda outside of the function call and this will make it much nicer to read:

```Kotlin
featureSync.onSyncRunning(this) {
    launch {
        val txtView: TextView = findViewById(R.id.historySyncStatus)
        txtView.text = getString(R.string.syncing)
    }
}
```

Finally, Kotlin Android Extensions are enabled for this sample app and you do not need to find the view yourself. The extensions will generate the code for you. Just import the property with the same id from (in this case) `kotlinx.android.synthetic.main.activity_main.`. Android Studio should do that automatically for you once you start typing and autocompleting `historySyncStatus`.

```Kotlin
featureSync.onSyncRunning(this) {
    launch {
        historySyncStatus?.text = getString(R.string.syncing)
     }
}
```

I also wonder whether you want to explicitly use the Main dispatcher here to run this on the UI thread. Afaik that's not guaranteed here.